jenkins-design-language/src/js/components/forms/TextInput.jsx

```diff
@@ -46,6 +47,7 @@ TextInput.propTypes = {
     iconRight: PropTypes.string,
     onChange: PropTypes.func,
     onBlur: PropTypes.func,
+    ariaLabel: PropTypes.string
```

Comment:
Why not stick with `aria-label`? Doing it this way means we're using two different prop names depending on whose component we're using.

Reply:
other ones aren't snake-case tho?

Reply:
What I mean is in your calling render you'd have `aria-label` on html elements or other-people's components, and `ariaLabel` on our TextInput component.

Reply:
@sophistifunk that is more or less the "react-way" to solve the `-` problem. 

You cannot do `const { ariaLabel = this.props.placeholder } = this.props;` anymore because this.props now contains `aria-label` and you cannot do ` const {aria-label}` so that needs to become `const ariaLabel = this.props['aria-label'] || this.props.placeholder;`

I changed it now 
![thingdoforlove](https://user-images.githubusercontent.com/596701/30903757-21085200-a370-11e7-8835-f4c414e99007.gif)

Reply:
@sophistifunk Actually the preferred way is to have something like

TextInput.propTypes = {
    'aria-*: PropTypes.string
}

...and allow arbitrary `aria` props, but I do not know how to do that.

Reply:
Yes a general solution would be better, next time we see this kind of thing we should cook up a little helper func. There is definitely documentation recommending not renaming `data-foo` to `dataFoo` though, and I'm assuming it holds for `aria-foo`.

Reply:
I think it's just something we're going to have eat, since W3C likes hyphens in names and ECMA does not :)

Reply:
@sophistifunk +1